src/dashboard/infrastructure/analytics-4/site-kit-analytics-4-adapter.php

```diff
@@ -41,31 +43,65 @@ public function __construct() {
+	// phpcs:disable VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable -- Temporary issue until this function gets an actual implementation.
+
 	/**
-	 * The wrapper method to do a Site Kit API request for Analytics 4.
+	 * The wrapper method to do a normal Site Kit API request for Analytics 4.
+	 * Added here to satisfy the contract of the interface. Will be filled once we have an implementation for it.
 	 *
-	 * @param Analytics_4_Parameters $parameters The parameters.
+	 * @param Parameters $parameters The parameters.
+	 *
+	 * @return Data_Container The Site Kit API response.
+	 *
+	 * @throws Invalid_Request_Exception This is an invalid request.
+	 */
+	public function get_data( Parameters $parameters ): Data_Container {
+		throw new Invalid_Request_Exception( 'No implementation of simple get data requests yet' );
+	}
+
+	// phpcs:enable
+
+	/**
+	 * The wrapper method to do a comparison Site Kit API request for Analytics.
+	 *
+	 * @param Parameters $parameters The parameters.
 	 *
 	 * @return Data_Container The Site Kit API response.
 	 *
-	 * @throws Failed_Request_Exception When the request responds with an error from Site Kit.
+	 * @throws Failed_Request_Exception      When the request responds with an error from Site Kit.
 	 * @throws Unexpected_Response_Exception When the request responds with an unexpected format.
+	 * @throws Invalid_Request_Exception     When the request is invalid due to unexpected parameters.
 	 */
-	public function get_data( Analytics_4_Parameters $parameters ): Data_Container {
+	public function get_comparison_data( Parameters $parameters ): Data_Container {
 		$api_parameters = $this->build_parameters( $parameters );
-		$response       = self::$analytics_4_module->get_data( 'report', $api_parameters );
 
-		if ( \is_wp_error( $response ) ) {
-			$error_data        = $response->get_error_data();
-			$error_status_code = ( $error_data['status'] ?? 500 );
-			throw new Failed_Request_Exception( \wp_kses_post( $response->get_error_message() ), (int) $error_status_code );
-		}
+		$response = self::$analytics_4_module->get_data( 'report', $api_parameters );
 
-		if ( ! \is_a( $response, RunReportResponse::class ) ) {
-			throw new Unexpected_Response_Exception();
-		}
+		$this->validate_response( $response );
 
-		return $this->parse_response( $response );
+		return $this->parse_comparison_response( $response );
+	}
+
+	/**
+	 * The wrapper method to do a daily Site Kit API request for Analytics.
+	 *
+	 * @param Parameters $parameters The parameters.
+	 *
+	 * @return Data_Container The Site Kit API response.
+	 *
+	 * @throws Failed_Request_Exception      When the request responds with an error from Site Kit.
+	 * @throws Unexpected_Response_Exception When the request responds with an unexpected format.
+	 * @throws Invalid_Request_Exception     When the request is invalid due to unexpected parameters.
+	 */
+	public function get_daily_data( Parameters $parameters ): Data_Container {
```

Comment:
I am not sure about the interface since now you need to use the generic `Parameters` here but we always rely on a specific implementation of this parameter object in these adapters. In this implementation it would be valid and we are signaling this as well that this could still work with the Search console parameters. Which is not the case. I think we need to revisit this idea and come up with something smart (together or not)

Reply:
We can keep the interface and type hint the specific parameters in the doc block:
```
* @param Analytics_4_Parameters $parameters The parameters.
```
If we really want to guard against the specific parameters, we can also manually do that in the function.

Otherwise, we can just remove the `Dashboard_Adapter_Interface` interface and have the adapters be totally separate classes.

Reply:
I removed the interface instead, [here](https://github.com/Yoast/wordpress-seo/pull/22067/commits/7be4bf31cffddfbb8168d30e96dff75e39d5a9f6).